### PR TITLE
change scripts/docs to generate one docs site per major.minor

### LINKED ISSUE #2409: Move to a single docs site per major.minor version
**Describe the problem/challenge you have**
Currently, we typically publish a new versioned docs site for each major.minor.patch release. This is somewhat burdensome from a maintenance perspective because there are more copies of the docs to update in case of change.

**Describe the solution you'd like**
GIven that patch releases should be bug-fix only, we'd like to maintain a single versioned docs site for a major.minor series.

### PR DESCRIPTION
**Pull request by skriss**
Signed-off-by: Steve Kriss <krisss@vmware.com>

closes #2409 

I did some basic verification on this by generating a `v1.4-pre` docs site and checked links, etc. It all looks good. We'll beta-test this process again when we ship `v1.4.0-beta.1`.

### CODE REVIEW

File: hack/gen-docs.sh [hunk 1]
```diff
@@ -66,8 +66,8 @@ rm site/_data/$NEW_DOCS_TOC.yml && cp site/_data/master-toc.yml site/_data/$NEW_
 # so check which OS we're running on.
 if [[ $(uname) == "Darwin" ]]; then
     echo "[OS X] updating version-specific links"
-    find site/docs/${NEW_DOCS_VERSION} -type f -name "*.md" | xargs sed -i '' "s|https://velero.io/docs/master|https://velero.io/docs/$NEW_DOCS_VERSION|g"
-    find site/docs/${NEW_DOCS_VERSION} -type f -name "*.md" | xargs sed -i '' "s|https://github.com/vmware-tanzu/velero/blob/master|https://github.com/vmware-tanzu/velero/blob/$NEW_DOCS_VERSION|g"
+    find site/docs/${NEW_DOCS_VERSION} -type f -name "*.md" | xargs sed -i '' "s|https://velero.io/docs/master|https://velero.io/docs/$VELERO_VERSION|g"
```
Comment (ashish-amarnath) [inline hack/gen-docs.sh:69]:
```suggestion
    find site/docs/${VELERO_VERSION} -type f -name "*.md" | xargs sed -i '' "s|https://velero.io/docs/master|https://velero.io/docs/$VELERO_VERSION|g"
```

Comment (ashish-amarnath) [reply hack/gen-docs.sh:69]: actually nvm.

File: site/docs/master/release-instructions.md [hunk 2]
```diff
@@ -17,12 +17,15 @@ You also need to update `site/index.html` to have "Latest Release Information" c
...
-1.  If there is an existing set of pre-release versioned docs for the version you are releasing (i.e. `site/docs/v1.2.0-beta.1` exists, and you're releasing `v1.2.0-beta.2` or `v1.2.0`):
+1.  If there is an existing set of pre-release versioned docs for the version you are releasing (i.e. `site/docs/v1.4-pre` exists, and you're releasing `v1.4.0-beta.2` or `v1.4.0`):
     - Remove the directory containing the pre-release docs, i.e. `site/docs/<pre-release-version>`.
     - Delete the pre-release docs table of contents file, i.e. `site/_data/<pre-release-version>-toc.yml`.
     - Remove the pre-release docs table of contents mapping entry from `site/_data/toc-mapping.yml`.
     - Remove all references to the pre-release docs from `site/_config.yml`.
-1.  Run `NEW_DOCS_VERSION=<VERSION> make gen-docs` (e.g. `NEW_DOCS_VERSION=v1.2.0 make gen-docs` or `NEW_DOCS_VERSION=v1.2.0-beta.1 make gen-docs`).
+1.  Run `NEW_DOCS_VERSION=<vMAJOR.MINOR> VELERO_VERSION=<version> make gen-docs` (e.g. `NEW_DOCS_VERSION=v1.2 VELERO_VERSION=v1.2.0 make gen-docs` or `NEW_DOCS_VERSION=v1.2-pre VELERO_VERSION=v1.2.0-beta.1 make gen-docs`).
```
Comment (ashish-amarnath) [inline site/docs/master/release-instructions.md:25]:
```suggestion
1.  Run `NEW_DOCS_VERSION=v<MAJOR.MINOR> VELERO_VERSION=v<version> make gen-docs` (e.g. `NEW_DOCS_VERSION=v1.2 VELERO_VERSION=v1.2.0 make gen-docs` or `NEW_DOCS_VERSION=v1.2-pre VELERO_VERSION=v1.2.0-beta.1 make gen-docs`).
```

Comment (skriss) [reply site/docs/master/release-instructions.md]: changed!

File: Makefile [hunk 3]
```diff
@@ -290,7 +290,7 @@ serve-docs:
 # tagged version. Once the unstaged changes are ready, they can be added to the
 # staging area using 'git add' and then committed.
 #
-# To run gen-docs: "NEW_DOCS_VERSION=v1.1.0 make gen-docs"
+# To run gen-docs: "NEW_DOCS_VERSION=v1.4 VELERO_VERSION=v1.4.0 make gen-docs"
```
Comment (ashish-amarnath) [inline Makefile:293]: This PR is to allow us to reuse the same docs version across patch releases, correct?
In which case, I think we should be updating the `gen-docs.sh` at https://github.com/vmware-tanzu/velero/pull/2473/files#diff-5f45d33e4fc9e049e1b0beee45db8269R25, no?

Comment (ashish-amarnath) [reply Makefile:293]: This link should be better https://github.com/vmware-tanzu/velero/blob/master/hack/gen-docs.sh#L25

Comment (skriss) [reply Makefile:293]: Well, you shouldn't really have to run the gen-docs process more than once per major/minor -- any "bug fixes" to the docs accompanying a patch release would be made manually in the major/minor.  Additionally, you could run into issues where docs changes made in master that are intended to be for the next feature release would get automatically pulled into a patch release's docs.

Comment (ashish-amarnath) [reply Makefile:293]: That sounds like what we've been doing in practice and also reasonable to continue. So 👍

**Pull request merged**